R/getPrevalence.R

```diff
@@ -300,17 +346,81 @@ setMethod("getRareTaxa", signature = c(x = "SummarizedExperiment"),
     function(x, rank = NULL, ...){
         # Gets the prevalent taxa
         prev_taxa <- getPrevalentTaxa(x, rank = rank, ...)
+        # If agglomeration was done
         if( !is.null(rank) ){
             # Gets names from specified taxonomic level
             taxa <- rowData(x)[[rank]]
-        } else{
-            # Gets rownames if agglomeration is not done
-            taxa <- rownames(x)
+        } else {
+            # If rownames is not NULL
+            if( !is.null(rownames(x)) ){
+                # Gets rownames 
+                taxa <- rownames(x)
+            } else {
+                # Gets indices of taxa
+                taxa <- 1:nrow(tse)
```

Comment:
This will be returned as warning by `BiocCheck`. Never ever use `1:x`, always use `seq_along` or `seq_len`

Reply:
Changed